src/OSPSuite.Infrastructure.Import/Core/Mappers/DataSetToDataRepositoryMapper.cs

```diff
@@ -59,12 +60,13 @@ public DataRepository ConvertImportDataSet(ImportedDataSet dataSet)
       private void convertParsedDataColumn(DataRepository dataRepository, KeyValuePair<ExtendedColumn, IList<SimulationPoint>> column, string fileName)
       {
          DataColumn dataColumn;
-         //so the dimension is calculated correctly but with the wrong function
-         //for now it stays like this, cause otherwise we have to have the association
-         //var dimension_test = _dimensionFactory.Dimension(column.Key.Name);
-         //otherwise we can also just get the defaultDimension from the columnInfos when we
-         //create the extended column
-         var dimension = _dimensionFactory.DimensionForUnit(column.Key.Column.Unit.SelectedUnit) ?? Constants.Dimension.NO_DIMENSION;
+
+         IDimension dimension;
+         if (column.Key.Column.Dimension != null)
+            dimension = column.Key.Column.Dimension;
+         else
```

Comment:
what about you create a varialbe for `column.Key.Column.Unit.SelectedUnit` and use it?
(you actgualoly create one later on).

This syntax column.key.Column is very strange by the way...

Reply:
column renamed to columnAndData (it is a cache of column variables like Dimension and a list of simulation points)

Reply:
variable "unit" intriduced earlier on